src/app/todos/todo.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+
+import { Todo, TodoStatus } from './todo.model';
+
+@Injectable()
+export class TodoService {
+  todosChanged = new Subject<Todo[]>();
+
+  private todos: Todo[] = [
+    new Todo( '1', 'Install Angular', TodoStatus.Done),
+    new Todo( '2', 'Add Edit TODO title', TodoStatus.Done),
+    new Todo( '3', 'Create basic TODO app', TodoStatus.InProgress),
+    new Todo( '4', 'Change UI from Bootstrap to Angular material', TodoStatus.Backlog),
+    new Todo( '5', 'Make UI like microsoft TODO app', TodoStatus.Backlog),
+    new Todo( '6', 'Add Edit TODO title', TodoStatus.Backlog),
+    new Todo( '7', 'Add Firebase API', TodoStatus.Backlog),
+    new Todo( '8', 'Add Drag-n-drop status columns', TodoStatus.Backlog),
+  ];
+
+  constructor() {}
+
+  // @ts-ignore
+  getTodo(id: string): Todo {
+    const todo = this.todos.find(todo => todo.id === id);
+
+    if (todo) {
+      return todo;
+    }
+  }
```

Comment:
Return type of the function is Todo, however, if there is no todo, looks like an undefined is returned? Looking at that `if` statement, you should ubdate a return type and add `else` as this function needs to return something